pkg/private/make_starlark_library.bzl

```diff
@@ -9,13 +9,10 @@ def _make_starlark_library(ctx):
     direct = []
     transitive = []
     for src in ctx.attr.srcs:
-        if StarlarkLibraryInfo in src:
```

Comment:
No; in fact I can drop this commit from this particular pull request.

I just find that it is incorrect here because 

```
transitive.append(src[StarlarkLibraryInfo])
```

This is appending the `StarlarkLibraryInfo`, not the file, to the transitive list, causing the depset creation below to fail when there is a bzl_library/starlark_library in `srcs`. I found this bug when I was trying to add rules_python/skylib to `//pkg:bzl_srcs` directly. But later I didn't actually add rules_python/skylib to `//pkg:bzl_srcs`.

Do you want me to drop this commit from this pull request and create a separate pull request?

Reply:
In other words, if I were to use `starlark_library` instead of `bzl_library` for the newly added `//pkg:rules_pkg_lib` target, then I do need this patch (or even fix it properly). Though, I think `@bazel_skylib//lib:paths` and `@rules_python//python:defs_bzl` are both `bzl_library`s, and due to this:

https://github.com/bazelbuild/bazel-skylib/blob/56b235e700ddd6a15b7d9fa1803fa7a84048471e/rules/private/bzl_library.bzl#L37

It is actually okay to just use the default files from these libraries without poking into StarlarkLibraryInfo, so this branch could potentially be deleted. 

Though, a pedantic person may also argue that, for this to be 100% correct, we should use the fields in StarlarkLibraryInfo from dependencies to construct StarlarkLibraryInfo of this starlark_library...

Reply:
Based on https://github.com/bazelbuild/bazel/issues/18391#issuecomment-2445364390

it seems we can drop the use of starlarklibrary and bzl_library.

It seem to work for me by changing pkg/BUILD:bzl_srcs to a filegroup(name ="rules_pkg_lib"
and referring to that in doc_build/BUILD, deleting the existing rules_pkg_lib there.

Reply:
> changing pkg/BUILD:bzl_srcs to a filegroup(name ="rules_pkg_lib"

bzl_srcs has visibility public. Wouldn't changing it to a filegroup breaks an API?

I can push a change that does this.

Reply:
Also, as I said in the original commit message:

> This is because the
//doc_build package is not loadable when rules_pkg itself
is not the root Bazel module, because stardoc is a
dev_dependency.

So how should users of rules_pkg build stardoc for a `.bzl` that relies on a .bzl from rules_pkg? It seems to me that the `bzl_library` must be outside of `//docs_build`. I choose `//pkg` because it is next to the .bzl files, which is what the other projects usually do as well, e.g. rules_cc does this:

https://github.com/bazelbuild/rules_cc/blob/34f0e1f038cff9bf28d65d101c806b48f35bf92e/cc/BUILD#L85

https://github.com/bazelbuild/rules_cc/blob/34f0e1f038cff9bf28d65d101c806b48f35bf92e/docs/BUILD#L30